### PR TITLE
Refactor `TraceOodFrame`

### PR DESCRIPTION
**Pull request by plafer**
Applies https://github.com/facebook/winterfell/pull/247#discussion_r1537123827

### CODE REVIEW

**Review by irakliyk — APPROVED** [review]
Looks good! Thank you! I left just one small nit comment inline.

File: air/src/proof/ood_frame.rs [hunk 1]
```diff
@@ -50,44 +44,52 @@ impl OodFrame {
     /// rows interleaved. This is done so that reseeding of the random coin needs to be done only
     /// once as opposed to once per each row.
     ///
+    /// The out-of-domain frame is stored as one vector of interleaved values, one from the
+    /// current row and the other from the next row. Given the input frame
```
[the remark this answers is not in the document]
Comment (plafer) [reply air/src/proof/ood_frame.rs]: fixed

File: air/src/proof/ood_frame.rs [hunk 2]
```diff
@@ -45,49 +39,53 @@ impl OodFrame {
...
+    /// trace states.
+    ///
+    /// The out-of-domain frame is stored as one vector of interleaved values, one from the current
+    /// row and the other from the next row. Given the input frame
+    ///
+    ///    +-------+-------+-------+-------+-------+-------+-------+-------+ |  a1   |   a2  |  ...
+    ///    |   an  |  c1   |  c2   |  ...  |  cm   |
+    ///    +-------+-------+-------+-------+-------+-------+-------+-------+ |  b1   |   b2  |  ...
+    ///    |   bn  |  d1   |  d2   |  ...  |  dm   |
+    ///    +-------+-------+-------+-------+-------+-------+-------+-------+
```
Comment (irakliyk) [inline air/src/proof/ood_frame.rs:52]: I think the formatting here is a bit off.

Comment (plafer) [reply air/src/proof/ood_frame.rs]: oops, fixed

**Review by irakliyk — APPROVED** [review]
Looks good! Thank you!

**Pull request merged**